### PR TITLE
fix: automatically link user without prompt

### PR DESCRIPTION
**Pull request by livio-a**
# Which Problems Are Solved

There were UX issue with the autolinking prompt page and users were not able to link their account or would not understand what to do. Since the trust to the IdP is already bound by the configuration, the user can directly be linked without any user input.

# How the Problems Are Solved

- remove the prompt page and directly link the user if possible
- remove corresponding customization texts from the API and Console

# Additional Changes

None

# Additional Context

- relates to https://github.com/zitadel/zitadel/issues/7977
- discussed with customers
- created as a `fix` to be able to backport

### CODE REVIEW

File: internal/api/ui/login/external_provider_handler.go [hunk 1]
```diff
@@ -501,10 +501,22 @@ func (l *Login) checkAutoLinking(w http.ResponseWriter, r *http.Request, authReq
...
 		return false
 	}
-	l.renderLinkingUserPrompt(w, r, authReq, user, nil)
+	l.autoLinkUser(w, r, authReq, user)
 	return true
 }
 
+func (l *Login) autoLinkUser(w http.ResponseWriter, r *http.Request, authReq *domain.AuthRequest, user *query.NotifyUser) {
+	err := l.authRepo.SelectUser(r.Context(), authReq.ID, user.ID, authReq.AgentID)
+	if err != nil {
```
Comment (stebenz) [inline internal/api/ui/login/external_provider_handler.go:510]: Should we make a one-liner out of the error condition?

File: internal/api/ui/login/external_provider_handler.go [hunk 2]
```diff
@@ -501,10 +501,22 @@ func (l *Login) checkAutoLinking(w http.ResponseWriter, r *http.Request, authReq
...
+		l.renderError(w, r, authReq, err)
+	}
+	err = l.authRepo.LinkExternalUsers(r.Context(), authReq.ID, authReq.AgentID, domain.BrowserInfoFromRequest(r))
+	if err != nil {
```
Comment (stebenz) [inline internal/api/ui/login/external_provider_handler.go:514]: Same here, should we make a one-liner out of the error condition?

File: internal/api/ui/login/external_provider_handler.go (hunk 1 above)
Comment (livio-a) [reply internal/api/ui/login/external_provider_handler.go:510]: more importantly i added a `return` on error

Comment (stebenz) [reply internal/api/ui/login/external_provider_handler.go]: Good point, missed that as the renderError function was there.

**Review by stebenz — APPROVED** [review]

**Pull request merged**